### DISCUSSION TITLE
Version 5.1

### DISCUSSION THREAD

**Discussion by stockiNail** [pepstock-org/Charba#68] [General]
**5.1** is the last stable release.

Documentation: [https://pepstock-org.github.io/Charba-Wiki](https://pepstock-org.github.io/Charba-Wiki).
API JavaDoc: [https://pepstock-org.github.io/Charba/5.1](https://pepstock-org.github.io/Charba/5.1).
Showcase GWT: [https://pepstock-org.github.io/Charba-Showcase](https://pepstock-org.github.io/Charba-Showcase).
Showcase J2CL: [https://pepstock-org.github.io/Charba-Showcase-J2CL](https://pepstock-org.github.io/Charba-Showcase-J2CL).

# Breaking changes
 
 * change `PointStyleCallback` class, adding context as generic.
 * remove `xPadding` and `yPadding` options from `LineLabel` options. Use padding instead.
 * remove `onDestroy` method from `Plugin` interface. Use `onBeforeDestroy` instead.
 * change type to `mode` and `overScaleMode` options in Zoom plugin configuration, from `org.pepstock.charba.client.enums.InteractionAxis` to `org.pepstock.charba.client.zoom.enums.Mode` class.
 * change signature of `EnterCallback`, `LeaveCallback`, `ClickCallback` and `DoubleClickCallback` interfaces of annotation plugin in order to get an `AnnotationContext` as argument instead of a chart and annotation options instances.
 
# Features

 * import CHART.JS version [v3.7.1](https://github.com/chartjs/Chart.js/releases/tag/v3.7.1).
 * import CHART.JS TREEMAP controller version [v2.0.2](https://github.com/kurkle/chartjs-chart-treemap/releases/tag/v2.0.2).
 * import CHART.JS MATRIX controller version [v1.1.1](https://github.com/kurkle/chartjs-chart-matrix/releases/tag/v1.1.1).
 * import CHART.JS SANKEY controller version [v0.8.2](https://github.com/kurkle/chartjs-chart-sankey/releases/tag/v0.8.1).
 * import CHART.JS GEO controller version [v3.7.1](https://github.com/sgratzl/chartjs-chart-geo/releases/tag/v3.7.1).
 * import CHART.JS ANNOTATION plugin version [v1.3.1](https://github.com/chartjs/chartjs-plugin-annotation/releases/tag/v1.3.1).
 * import LUXON library version [v2.3.0](https://github.com/moment/luxon/releases/tag/2.3.0).
 * add `clip` property to annotation options.
 * add `pointStyle`, `rotation`, `xAdjust` and `yAdjust` properties to `PointAnnotation` options.
 * add new `PolygonAnnotation` class in order to draw a polygon.
 * add new `LabelAnnotation` class in order to draw a content (text, image or canvas) on the chart.
 * add point positioning by box options for `PointAnnotation` class.
 * add label options to `BoxAnnotation` class. 
 * enable to set a content by `Canvas` instance to `LineLabel` options.
 * enable to set the `position` of the `LineLabel` options as percentage of the line dimension.
 * add `borderShadowColor`, `backgroundShadowColor`, `shadowBlur`, `shadowOffsetX` and `shadowOffsetY` to annotations and line label options in order to enable the shadowing.
 * add `arrowHeads` option to `LineAnnotation` class options in order to enable the arrows drawing to the line.
 * add the annotation element to the `AnnotationContext` in order to pass to the user the dimension and options of annotation (i.e for gradient).
 * add `drawActiveElementsOnTop` property to lining and bubble datasets.
 * add `onBeforeDestroy` and `onAfterDestroy` methods to `Plugin` interface.
 * enable all tick callbacks to return a `List` of strings in order to have ticks multiple lines.
 * enable `PointLabelCallback` class to return a `List` of strings in order to have ticks multiple lines.
 * add `inCharArea` option to `PluginEventArgument` class.
 * add `r` item to `InteractionAxis` class.
 * add `centered` option to `PointLabels` and `RadialPointLabels` classes.
 * add `borderJoinStyle` option to `Arc` element options and configuration, to `PolarAreaDataset`, `PieDataset` and `DoughnutDataset` classes.
 * add `padding` option to `ProjectionAxis` class.
 * add `changed` option to `PluginEventArgument` class in order to enable the rendering of the chart after event processing.
 * add a `IsPoint` instance argument to the `setTooltipActiveElements` method in the chart.  

# Developing

 * change dependency for Google Closure Compiler, version [v20220202](https://mvnrepository.com/artifact/com.google.javascript/closure-compiler/v20220202).

<hr /><em>This discussion was created from the release <a href='https://github.com/pepstock-org/Charba/releases/tag/5.1'>Version 5.1</a>.</em>

**Comment by salmonb** [comment]
Thanks for your work.

I'm bumping Charba from version 3.3-gwt to 5.1-gwt. After resolving the breaking changes, the app works fine but the area in the StackedAreaChart has no color anymore even if I call setBackgroundColor() on the dataset. Is it because it's not anymore the correct way to set the area color, or is it a bug? (I can open an issue in that case if you prefer).

Thank you.

**Reply by stockiNail** [reply]
@salmonb have a look to issue https://github.com/pepstock-org/Charba/issues/69. because I think I found why you have that behavior.

**Reply by salmonb** [reply]
I confirm you found the reason. You can close the issue. Thank you very much!

**Comment by stockiNail** [comment]
Thank you very much for your feedback.

I have opened a issue #69.. :(
